nixos/modules/system/activation/switch-to-configuration.pl

```diff
@@ -136,358 +107,395 @@ sub fingerprintUnit {
     return abs_path($s) . (-f "${s}.d/overrides.conf" ? " " . abs_path "${s}.d/overrides.conf" : "");
 }
 
-# Figure out what units need to be stopped, started, restarted or reloaded.
-my (%unitsToStop, %unitsToSkip, %unitsToStart, %unitsToRestart, %unitsToReload);
+sub main {
+    my $action = shift @ARGV;
```

Comment:
To avoid using the global variable `@ARGV` inside main

```suggestion
    my $action = shift;
```